src/django/api/models.py

```diff
@@ -2659,6 +2659,43 @@ class ArraySubquery(Subquery):
     template = 'ARRAY(%(subquery)s)'
 
 
+def get_sector_dict(facility_ids):
+    item_sectors = FacilityMatch \
+        .objects \
+        .filter(facility_id=OuterRef('pk')) \
+        .filter(Q(is_active=True)
+                & Q(status__in=[FacilityMatch.AUTOMATIC,
+                                FacilityMatch.CONFIRMED,
+                                FacilityMatch.MERGED])) \
+        .order_by('facility_list_item__source__contributor_id',
+                  '-facility_list_item__source__is_active',
+                  '-updated_at') \
+        .annotate(values=Func('facility_list_item__sector',
+                              function='unnest')) \
+        .distinct('facility_list_item__source__contributor_id') \
```

Comment:
We didn't previously do this `distinct('...__contributor_id')` in `index_facilities` - instead we did `distinct` over the full unnested list of sectors for each facility. Are we OK dropping any active matches for this contributor that aren't the most recent?

Reply:
Good question. We keep immutable history of what was submitted in our FacilityListItem table so we can always get back to it if wee need it. Indexing by only the most recent values submitted by each contributor allows the contributor to remove a mistakenly submitted value and ensures that mistake is not permanently searchable. If, over the course of the beta period the OS Hub team gets feedback that the old values should remain searchable for some reason, we can rework this.